modules/dists/BlockDist.chpl

```diff
@@ -1536,12 +1536,20 @@ where this.sparseLayoutType == unmanaged DefaultDist &&
 }
 
 // Block1 <=> Block2 
-proc BlockArr.doiSwap(arr) {
-  coforall (locarr1, locarr2) in zip(this.locArr, arr.locArr) {
-    on locarr1 {
-      locarr1.myElems <=> locarr2.myElems;
-      locarr1.locRAD <=> locarr2.locRAD;
+proc BlockArr.doiOptimizedSwap(other) {
+  if(this.dom.dist == other.dom.dist &&
```

Comment:
There isn't much need for this first check. If you were to keep it, you can change the logic to use `||`